csharp/src/Microsoft.ML.OnnxRuntime/CheckpointState.shared.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    ///  Sets various runtime options. 
+    /// </summary>
+    public class CheckpointState : SafeHandle
+    {
+        internal IntPtr Handle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        /// <summary>
+        /// Default __ctor. Creates default RuntimeOptions
+        /// </summary>
+        public CheckpointState()
+            : base(IntPtr.Zero, true)
+        {
+            var envHandle = OrtEnv.Handle; // just so it is initialized
+        }
```

Comment:
The usage pattern is not clear. Can we use default constructed CheckPoint? It seems useless, but the client code would still want to call Dispose() on it. Encourages a useless pattern.
Would we rather want to make LoadCheckout() a static factory method?

Reply:
done